sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -110,10 +110,6 @@ object TableOutputResolver {
       resolveColumnsByPosition(tableName, query.output, actualExpectedCols, conf, errors += _)
     }
 
-    if (errors.nonEmpty) {
```

Comment:
I would jus tthrow an exception here since this is a followup of SPARK-42309. If we need to fix a different issue, we should have another jira.

Reply:
Make sense to me. Created new JIRA SPARK-45476 and update the PR title & description.